analyzers/src/SonarAnalyzer.CSharp/Rules/RedundantInheritanceList.cs

```diff
@@ -145,50 +124,45 @@ private static void ReportRedundantInterfaces(SyntaxNodeAnalysisContext context,
                     collidingDeclaration.ToMinimalDisplayString(context.SemanticModel, baseType.Type.SpanStart),
                     interfaceType.ToMinimalDisplayString(context.SemanticModel, baseType.Type.SpanStart));
 
-                context.ReportIssue(Diagnostic.Create(Rule, location,
-                    ImmutableDictionary<string, string>.Empty.Add(RedundantIndexKey, i.ToString(CultureInfo.InvariantCulture)),
-                    message));
+                context.ReportIssue(Diagnostic.Create(Rule, location, DiagnosticsProperties(redundantIndex: i), message));
             }
         }
 
-        private static MultiValueDictionary<INamedTypeSymbol, INamedTypeSymbol> GetImplementedInterfaceMappings(BaseListSyntax baseList, SemanticModel semanticModel) =>
+        private static ILookup<INamedTypeSymbol, INamedTypeSymbol> GetImplementedInterfaceMappings(BaseListSyntax baseList, SemanticModel semanticModel) =>
             baseList.Types
                     .Select(baseType => semanticModel.GetSymbolInfo(baseType.Type).Symbol as INamedTypeSymbol)
                     .WhereNotNull()
                     .Distinct()
-                    .Select(symbol => new Tuple<INamedTypeSymbol, ICollection<INamedTypeSymbol>>(symbol, symbol.AllInterfaces))
-                    .ToMultiValueDictionary(kv => kv.Item1, kv => kv.Item2);
+                    .ToLookup(x => x.AllInterfaces.AsEnumerable());
 
         private static bool TryGetCollidingDeclaration(INamedTypeSymbol declaredType,
                                                        INamedTypeSymbol interfaceType,
-                                                       MultiValueDictionary<INamedTypeSymbol, INamedTypeSymbol> interfaceMappings,
+                                                       ILookup<INamedTypeSymbol, INamedTypeSymbol> interfaceMappings,
                                                        out INamedTypeSymbol collidingDeclaration)
         {
-            var collisionMapping = interfaceMappings
-                .Where(i => i.Key.IsInterface())
-                .FirstOrDefault(v => v.Value.Contains(interfaceType));
-
-            if (collisionMapping.Key != null)
+            var collisionMapping = interfaceMappings.FirstOrDefault(x => x.Key.IsInterface() && x.Contains(interfaceType));
```

Comment:
The `bool` result is redundant to `collidingDeclaration is { }` state. Let's change the signature to this single line:
```
private static INamedTypeSymbol CollidingDeclaration(INamedTypeSymbol declaredType, INamedTypeSymbol interfaceType, ILookup<INamedTypeSymbol, INamedTypeSymbol> interfaceMappings)
```
It will simplify the implementation, usage and state management in general.